src/test/groovy/edu/ie3/datamodel/models/input/thermal/CylindricalStorageInputTest.groovy

```diff
@@ -0,0 +1,38 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+package edu.ie3.datamodel.models.input.thermal
+
+import edu.ie3.test.common.ThermalUnitInputTestData
+import spock.lang.Specification
+
+
+class CylindricalStorageInputTest extends Specification {
+
+	def "A CylindricalStorageInput copy method should work as expected"() {
+		given:
+		def cylindricalStorageInput = ThermalUnitInputTestData.cylindricStorageInput
+
+		when:
+		def alteredUnit = cylindricalStorageInput.copy().storageVolumeLvl(ThermalUnitInputTestData.storageVolumeLvl)
+				.storageVolumeLvlMin(ThermalUnitInputTestData.storageVolumeLvlMin).inletTemp(ThermalUnitInputTestData.inletTemp)
+				.returnTemp(ThermalUnitInputTestData.returnTemp).c(ThermalUnitInputTestData.c).build()
```

Comment:
Test case for altered thermal bus is missing.

Reply:
Added.